lib/lightning_web/live/project_live/settings.ex

```diff
@@ -53,10 +90,23 @@ defmodule LightningWeb.ProjectLive.Settings do
        project_changeset: Projects.change_project(socket.assigns.project),
        can_delete_project: can_delete_project,
        can_edit_project_name: can_edit_project_name,
-       can_edit_project_description: can_edit_project_description
+       can_edit_project_description: can_edit_project_description,
+       show_github_setup: show_github_setup,
+       show_repo_setup: show_repo_setup,
+       show_sync_button: show_sync_button,
+       project_repo: project_repo,
+       repos: collect_project_repos(socket.assigns.project.id),
+       branches: []
      )}
   end
 
+  # we should only run this if repo setting is pending
+  defp collect_project_repos(project_id) do
+    {:ok, repos} = VersionControl.fetch_installation_repos(project_id)
```

Comment:
This is going to block the loading of the LiveView, rather make the repos assign `nil` (or something else) in `mount`, and something like:

```
pid = self()
Task.start(fn -> 
  {:ok, repos} = VersionControl.fetch_installation_repos(project_id)
  send(pid, {:repos_fetched, repos})
end)
```

Also consider error cases, and providing both loading and error feedback in the UI when this is in progress or has failed.